### ISSUE TITLE
Autosaved file is filled with zeros after system error

### ISSUE THREAD

**Issue by shams00** [jgraph/drawio-desktop#543]
**Is your feature request related to a problem? Please describe.**

Firstly thank you for the great piece of software. 
And now the problem. I have lost my diagram file after getting Windows BSOD (MEMORY_MANAGEMENT error).
I think drawio-desktop was autosaving when the BSOD happened.
The resulting diagram file after reboot was of size 8192 bytes and filled with 0x00.

**Describe the solution you'd like**
It would be great to have some kind of protection from system errors during autosave. For example, save to temporary file and then rename.

**Comment by davidjgraph** [comment]
What is the version under the help menu?

**Comment by shams00** [comment]
14.5.1

**Comment by davidjgraph** [comment]
There is a temporary file created that is deleted when the save is confirmed. If it's not there are you sure that's the version you were using when you saved?

**Comment by shams00** [comment]
I'm sure that I have being editing this file when the BSOD happend.

If you are telling about Draw.io version, I'm not sure that Draw.io version was 14.5.1 at the error moment because I haven't checked it. But I have found log [main.log](https://github.com/jgraph/drawio-desktop/files/6257213/main.log)
In my opinion there are some strange records in the log (multiple update to 14.5.1) so I will not interpret them by myself and just attach it.

My history in more details for correlation with the log:
- I have started Draw.io at 2021-04-04 17:18
- I have pressed "Update" (or something similar) in Draw.io autoupdater
- After the update has finished I have started the new diagram
- After few minutes I did "Save As" to the file
- Working with diagram
- BSOD at 2021-04-04 23:36 
- 23:42 "Error loading file. Not a diagram file" when trying to open diagram first time after BSOD

Right now there is no temporary file in this folder. But I have not checked for temporary file before trying to open the diagram right after the BSOD.
